### DISCUSSION TITLE
How can ESPResSo be utilized to solve the rotational motion equations of particles?

### DISCUSSION THREAD

**Discussion by ywem** [espressomd/espresso#5001] [General]
I have noticed that in tutorials and introductions related to Langevin dynamics, only the translational motion equations of particles are typically presented. Does ESPResSo have the capability to also solve for the rotational dynamics of particles?

**Comment by jngrad** [comment]
ESPResSo can solve rotation motion too. By default, particles are non-rotating. Rotation must be specified at particle creation (section [Rotational degrees of freedom and particle anisotropy](https://espressomd.github.io/doc/particles.html#rotational-degrees-of-freedom-and-particle-anisotropy)). Thermostats have a flag `gamma_rotation`. Anisotropy can be achieved by setting the particle rotational inertial tensor. The particle director, quaternion, and magnetic dipole are internally all linked together: modifying one modifies the other. Hope this helps! Sorry for the late answer.

**Comment by ywem** [comment]
Thans a lot for this explanation. It is really help.


Best wishes,


Wenming Yang









--

-------------------------------------------------------------------
Dr. Wenming Yang, Associate Professor
School of Mechanical Engineering
University of Science and Technology Beijing
Beijing, China, 100083
Tel: +86 13521498236
Email: ***@***.***; ***@***.***



At 2025-09-12 09:37:12, "Jean-Noël Grad" ***@***.***> wrote:

ESPResSo can solve rotation motion too. By default, particles are non-rotating. Rotation must be specified at particle creation (section Rotational degrees of freedom and particle anisotropy). Thermostats have a flag gamma_rotation. Anisotropy can be achieved by setting the particle rotational inertial tensor. The particle director, quaternion, and magnetic dipole are internally all linked together: modifying one modifies the other. Hope this helps! Sorry for the late answer.
